airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/spec/SpecTest.kt

```diff
@@ -80,7 +84,7 @@ abstract class SpecTest :
         val jsonMatcher: JsonMatcher =
             CompositeJsonMatcher(
                 StrictJsonArrayPartialMatcher(),
-                LenientJsonObjectPartialMatcher(),
+                StrictJsonObjectPartialMatcher(),
```

Comment:
previously this wasn't detecting new properties, switching to the strict matcher gives us the behavior we want